pkg/tasks/retry_task.go

```diff
@@ -0,0 +1,44 @@
+package tasks
+
+import (
+	"context"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+type RetryTaskConfig struct {
+	Logger     micrologger.Logger
+	Underlying Task
+}
+
+func NewRetryTask(config RetryTaskConfig) (Task, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Underlying == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Underlying must not be empty", config)
+	}
+
+	t := func(ctx context.Context) error {
+		o := func() error {
+			err := config.Underlying(ctx)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+		b := backoff.NewExponential(backoff.ShortMaxWait, backoff.ShortMaxInterval)
```

Comment:
Any reason why you choose `backoff.ShortMaxWait, backoff.ShortMaxInterval` in particular? Does it make sense to make it configurable at the Task level?

Reply:
No reason. That made sense for my particular use-case. If we need to configure that we can add that to config and default to the short.